src/Common/Core/Impl/Extensions/StringExtensions.cs

```diff
@@ -22,6 +23,43 @@ public static class StringExtensions {
         public static int LastIndexOfIgnoreCase(this string s, string searchFor, int startIndex) => s.LastIndexOf(searchFor, startIndex, StringComparison.OrdinalIgnoreCase);
         public static bool ContainsIgnoreCase(this string s, string prefix) => s.IndexOf(prefix, StringComparison.OrdinalIgnoreCase) >= 0;
 
+        public static IEnumerable<int> AllIndexesOfIgnoreCase(this string s, string value, int startIndex = 0, bool allowOverlap = false) {
+            var i = startIndex;
+            while (i < s.Length) {
+                i = s.IndexOf(value, i, StringComparison.OrdinalIgnoreCase);
+                if (i < 0) {
+                    break;
+                }
+
+                yield return i;
+                i = allowOverlap ? i + value.Length : i + 1;
+            }
+        }
+
+        public static bool IsStartOfNewLine(this string s, int index, bool ignoreWhitespaces = false) {
```

Comment:
There is a difference between those two methods for `position == 0`. `IsStartOfNewLine` will return `true`, while `IsNewLineBeforePosition` will return `false`.